lib/api/socket-handler.js

```diff
@@ -31,6 +32,15 @@ var SocketHandler = function(app, server, auth) {
   var self = this;
   this.io.sockets.on('connection', function(socket) {
     var clientQuery;
+    
+    // override the emit function to add logging
+    var emit = socket.emit;
+    socket.emit = function(event, data) {
+      logger.debug(event, data);
+      
+      socket.emit = emit;
+      socket.emit(event, data);
```

Comment:
On line 37, we store reference to original "emit", which is the actual meat and sends stuff back to browser.
On line 38, we create a new definition of "emit", where we log the data to be sent back.
On line 41, we reassign the original definition so that the original is called where ever needed from now on.
On line 42, we call the original to send data to the client